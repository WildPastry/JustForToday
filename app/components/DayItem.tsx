import { ColorSchemeName, Pressable, StyleSheet } from 'react-native';
import { IDate, IDayItem } from '../types/date.types';
import { AppState } from '../redux/store';
import Colours from '../constants/Colours';
import Control from '../constants/Control';
import { IDeviceSize } from '../types/generic.types';
import { Text } from './styles/Themed';
import getDeviceSize from '../constants/Layout';
import { useAppSelector } from '../redux/hooks';
import useColorScheme from '../hooks/useColorScheme';

const DayItem: React.FC<IDayItem> = (props: IDayItem): JSX.Element => {
  // Component settings
  const colorScheme: NonNullable<ColorSchemeName> = useColorScheme();
  const deviceSize: IDeviceSize[keyof IDeviceSize] = getDeviceSize();

  // Data from store
  const dates: IDate = useAppSelector((state: AppState): IDate => {
    return state.date;
  });

  const isCurrentDay = (): boolean => {
    return dates.currentDay === props.id;
  };

  return (
    <Pressable
      style={[
        styles.dayItem,
        isCurrentDay()
          ? { backgroundColor: Colours[colorScheme].currentBtn }
          : { backgroundColor: Colours[colorScheme].btn }
      ]}
      onPress={props.onPress}>
      <Text
        style={[
          styles.text,
          Control[deviceSize].text,
          isCurrentDay() ? { color: Colours[colorScheme].currentBtnText } : null
        ]}>
        {props.name}
      </Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'center'
  },
  dayItem: {
    borderRadius: 7,
    marginBottom: 15,
    paddingVertical: 10
  }
});

export default DayItem;
