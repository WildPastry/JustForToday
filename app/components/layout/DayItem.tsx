import { ColorSchemeName, Pressable, StyleSheet } from 'react-native';
import { IDate, IDayItem } from '../../types/date.types';
import { AppState } from '../../redux/store';
import { Text } from '../styles/Themed';
import itemStates from '../../constants/itemStates';
import { useAppSelector } from '../../redux/hooks';
import useColorScheme from '../../hooks/useColorScheme';

const DayItem: React.FC<IDayItem> = (props: IDayItem): JSX.Element => {
  // Component settings
  const colorScheme: NonNullable<ColorSchemeName> = useColorScheme();
  const currentDayTheme = itemStates[`${colorScheme}ActiveItem`];

  // Data from store
  const dates: IDate = useAppSelector((state: AppState): IDate => {
    return state.date;
  });

  const isCurrentDay = (): boolean => {
    return dates.currentDay === props.id;
  };

  return (
    <Pressable
      style={[
        styles.dayItem,
        isCurrentDay() ? currentDayTheme : itemStates[`${colorScheme}Item`]
      ]}
      onPress={props.onPress}>
      {/* Day item */}
      <Text style={[styles.text, isCurrentDay() ? styles.textWhite : null]}>
        {props.name}
      </Text>
    </Pressable>
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'center'
  },
  textWhite: {
    color: '#fff'
  },
  dayItem: {
    borderRadius: 8,
    marginBottom: 12,
    paddingVertical: 8
  }
});

export default DayItem;
