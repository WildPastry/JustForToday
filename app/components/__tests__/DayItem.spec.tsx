import DayItem from '../layout/DayItem';
import { Provider } from 'react-redux';
import { makeStore } from '../../redux/store';
import renderer from 'react-test-renderer';

describe('<DayItem />', () => {
  it('renders correctly', () => {
    const store = makeStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <DayItem id={''} name={''} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
