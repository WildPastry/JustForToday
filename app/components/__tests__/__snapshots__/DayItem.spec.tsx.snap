// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayItem /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 7,
        "marginBottom": 8,
        "paddingHorizontal": 6,
        "paddingVertical": 5,
      },
      {
        "backgroundColor": "#e5edf9",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "textAlign": "center",
        },
      ]
    }
  />
</View>
`;
